test/powershell/engine/PropertyAccessor.Tests.ps1

```diff
@@ -0,0 +1,96 @@
+    $IsNotSkipped = ($IsWindows -eq $true);
+    $PSDefaultParameterValues["it:skip"] = !$IsNotSkipped
+    $ShouldSkipTest = !$IsNotSkipped
+
+    Describe "User-Specific PowerShellProperties.json Modifications" -Tags "CI" {
+
+        BeforeAll {
+            if ($IsNotSkipped) {
+                # Discover the user-specific PowerShellProperties.json file
+                $splitModulePath = $env:PSMODULEPATH -split ";"
+                $userModuleDir = $splitModulePath[0].TrimEnd("\")
+                $userSettingsDir = $userModuleDir -replace "\\Modules", ""
+                $userPropertiesFile = Join-Path $userSettingsDir "PowerShellProperties.json"
+
+                # Save the file for restoration after the tests are complete
+                $backupPropertiesFile = ""
+                if (Test-Path $userPropertiesFile) {
+                    $backupPropertiesFile = Join-Path $userSettingsDir "ORIGINAL_PowerShellProperties.json"
+                    Copy-Item -Path $userPropertiesFile -Destination $backupPropertiesFile -Force -ErrorAction Continue
+                }
+                else {
+                    # create the directory and file if they do not already exist
+                    $null = New-Item -Type Directory  -Path $userSettingsDir -Force -ErrorAction SilentlyContinue
+                    Set-Content -Path $userPropertiesFile -Value '{"Microsoft.PowerShell:ExecutionPolicy":"RemoteSigned"}'
+                }
+
+                # Save original the Process ExecutionPolicy. The tests assume that it is Undefined
+                $processExecutionPolicy = Get-ExecutionPolicy -Scope Process
+                Set-ExecutionPolicy -Scope Process -ExecutionPolicy Undefined
+            }
+        }
+
+        BeforeEach {
+            if ($IsNotSkipped) {
+                Set-Content -Path $userPropertiesFile -Value '{"Microsoft.PowerShell:ExecutionPolicy":"RemoteSigned"}'
+            }
+        }
+
+        AfterAll {
+            if ($IsNotSkipped) {
+                if ([string]::IsNullOrEmpty($backupPropertiesFile))
+                {
+                    # Remove PowerShellProperties.json if it did not exist before the tests
+                    Remove-Item -Path $userPropertiesFile -Force -ErrorAction SilentlyContinue
+                }
+                else
+                {
+                    # Restore the original PowerShellProperties.json file if it existed before the test pass.
+                    Move-Item -Path $backupPropertiesFile -Destination $userPropertiesFile -Force -ErrorAction Continue
+                }
+
+                # Restore the original Process ExecutionPolicy
+                Set-ExecutionPolicy -Scope Process -ExecutionPolicy $processExecutionPolicy
+            }
+        }
+
+        It "Verify Queries to Missing File Return Default Value" {
+            Remove-Item $userPropertiesFile -Force
+
+            Get-ExecutionPolicy -Scope CurrentUser | Should Be "Undefined"
```

Comment:
Maybe add one more check to verify that read operation should not create the file.